src/glum/_glm.py

```
@@ -2474,3 +2478,172 @@ def fit(
+        warning for this case.
+
+        References
+        ----------
+        [1] Burnham KP, Anderson KR (2002). Model Selection and Multimodel
+        Inference; Springer New York.
+        [2] Zou, H., Hastie, T. and Tibshirani, R., (2007). On the “degrees of
+        freedom” of the lasso; The Annals of Statistics.
+        [3] Park, M.Y., 2006. Generalized linear models with regularization;
+        Stanford Universty.
+        """
+
+        # we require that the log_likelihood be defined
+        model_err_str = (
+            "The computation of the information criteria has only "
+            + "been defined for models with a Binomial likelihood or a Tweedie "
+            + "likelihood with power <= 2."
+        )
+        if not isinstance(
+            self.family_instance, (BinomialDistribution, TweedieDistribution)
+        ):
+            raise NotImplementedError(model_err_str)
+
+        # the log_likelihood has not been implemented for the InverseGaussianDistribution
+        if (
+            isinstance(self.family_instance, TweedieDistribution)
+            and self.family_instance.power > 2
+        ):
+            raise NotImplementedError(model_err_str)
+
+        ddof = np.sum(np.abs(self.coef_) > np.finfo(self.coef_.dtype).eps)
+        k_params = ddof + self.fit_intercept
+        nobs = X.shape[0]
+
+        if nobs != self._num_obs:
+            raise ValueError(
+                "The same dataset that was used for training should "
+                + "also be used for the computation of information "
+                + "criteria"
+            )
+
+        mu = self.predict(X)
+        ll = self.family_instance.log_likelihood(y, mu, sample_weight=sample_weight)
+
+        aic = -2 * ll + 2 * k_params
+        bic = -2 * ll + np.log(nobs) * k_params
+        if nobs > k_params + 1:
+            aicc = aic + 2 * k_params * (k_params + 1) / (nobs - k_params - 1)
+        else:
+            aicc = None
```

If aicc can be None, I would raise an error when someone tries to call it instead of returning None.